cmd/resultscollector/main.go

```diff
@@ -255,64 +263,103 @@ func main() {
+				if err != nil {
+					fmt.Println(err)
+					os.Exit(1)
+				}
+
+				var wg sync.WaitGroup
+				wg.Add(2)
+				go func() {
+					err = backoff.Retry(func() error {
+						url := scapresultsconf.ResultServerURI
+						fmt.Printf("Trying to upload to resultserver: %s\n", url)
+						reader := bytes.NewReader(arfContents.contents)
+						client := &http.Client{}
+						req, _ := http.NewRequest("POST", url, reader)
+						req.Header.Add("Content-Type", "application/xml")
+						req.Header.Add("X-Report-Name", scapresultsconf.ConfigMapName)
+						if arfContents.compressed {
+							req.Header.Add("Content-Encoding", "bzip2")
+						}
+						resp, err := client.Do(req)
+						if err != nil {
+							fmt.Println(err)
+							return err
+						}
+						defer resp.Body.Close()
+						bytesresp, err := httputil.DumpResponse(resp, true)
+						if err != nil {
+							fmt.Println(err)
+							return err
+						}
+						fmt.Println(string(bytesresp))
+						return nil
+					}, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), maxRetries))
 
-			var wg sync.WaitGroup
-			wg.Add(2)
-			go func() {
-				err = backoff.Retry(func() error {
-					url := scapresultsconf.ResultServerURI
-					fmt.Printf("Trying to upload to resultserver: %s\n", url)
-					reader := bytes.NewReader(arfContents.contents)
-					client := &http.Client{}
-					req, _ := http.NewRequest("POST", url, reader)
-					req.Header.Add("Content-Type", "application/xml")
-					req.Header.Add("X-Report-Name", scapresultsconf.ConfigMapName)
-					if arfContents.compressed {
-						req.Header.Add("Content-Encoding", "bzip2")
-					}
-					resp, err := client.Do(req)
 					if err != nil {
 						fmt.Println(err)
-						return err
+						os.Exit(1)
 					}
-					defer resp.Body.Close()
-					bytesresp, err := httputil.DumpResponse(resp, true)
+					fmt.Println("Uploaded to resultserver")
+					wg.Done()
+				}()
+
+				go func() {
+					err = backoff.Retry(func() error {
+						fmt.Println("Trying to upload ConfigMap")
+						openscapScan, err := getOpenSCAPScanInstance(scapresultsconf.ScanName, scapresultsconf.Namespace, dynclient)
+						if err != nil {
+							return err
+						}
+						confMap := getConfigMap(openscapScan, scapresultsconf.ConfigMapName, "results", xccdfContents.contents, xccdfContents.compressed, exitcode.contents)
+						_, err = clientset.CoreV1().ConfigMaps(scapresultsconf.Namespace).Create(confMap)
+
+						if errors.IsAlreadyExists(err) {
+							return nil
+						}
+						return err
+					}, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), maxRetries))
+
 					if err != nil {
 						fmt.Println(err)
-						return err
+						os.Exit(1)
 					}
-					fmt.Println(string(bytesresp))
-					return nil
-				}, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), maxRetries))
-
+					fmt.Println("Uploaded ConfigMap")
+					wg.Done()
+				}()
+				wg.Wait()
+			} else {
```

Comment:
The logic looks good, but instead of:
```
if ok {
 ok_stuff()
} else {
 error_stuff()
}
```
Could we do:
```
if !ok {
   error_stuff()
}
ok_stuff()
```
This is to keep the usual codepath to the left. Alternatively, if there is some shared code that needs to be executed in both cases, it would at least be nice to extract bits of code into functions. The current code is getting long and hard to read.